envoy-control-core/src/test/kotlin/pl/allegro/tech/servicemesh/envoycontrol/metrics/ThreadPoolMetricTest.kt

```diff
@@ -0,0 +1,35 @@
+package pl.allegro.tech.servicemesh.envoycontrol.metrics
+
+import io.micrometer.core.instrument.simple.SimpleMeterRegistry
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+import pl.allegro.tech.servicemesh.envoycontrol.ControlPlane
+import pl.allegro.tech.servicemesh.envoycontrol.EnvoyControlProperties
+import pl.allegro.tech.servicemesh.envoycontrol.server.ExecutorType
+import reactor.core.publisher.Flux
+
+class ThreadPoolMetricTest {
+
+    @Test
+    fun `should bind metrics for default executors`() {
+        // given
+        val meterRegistry = SimpleMeterRegistry()
+        val envoyControlProperties = EnvoyControlProperties().also {
+            it.server.executorGroup.type = ExecutorType.PARALLEL
+            it.server.groupSnapshotUpdateScheduler.type = ExecutorType.PARALLEL
+        }
+
+        val controlPlane = ControlPlane.builder(envoyControlProperties, meterRegistry).build(Flux.empty())
+
+        // when
+        controlPlane.start()
```

Comment:
Don't we need to stop at the end of the test? (otherwise we're just wasting resources).

Reply:
done